submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireService.kt

```diff
@@ -29,21 +33,47 @@ class FireService(
      * different path) and if so, even if the file exists in FIRE, it gets duplicated to ensure consistency.
      */
     fun getOrPersist(sub: ExtSubmission, file: ExtFile): FirePersistResult {
-        val expectedPath = "/${sub.relPath}/${file.relPath}"
```

Comment:
I had made this change because the paths are the same in both places which I think makes clear that no matter where the file is processed, we use the same path

Reply:
yes the problem is that this is not the path when file is a folder. As we need to add the .zip that is the reason I move the call after compressing the file and creating extDirectory. 

I did update the code to have single declaration please take a look again.